### PR TITLE
Feature/platform only policies

### PR DESCRIPTION
**Pull request by mikeurbanski1**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

With this change, by default, if using an API key, checkov will only run policies that exist in the platform (currently in the BC platform, regardless of API key type - because the results should match what they see in the platform). Platform policies are identified by the existence if `bc_id` on the check (in some cases, runners are responsible for setting this on their own checks, if they load them in a nonstandard way, like the secrets runner).

Also adds the flag `--include-all-checkov-policies` to include OOTB policies that only exist in checkov. Custom policies loaded locally will still run without this flag.

So, specifically:

* if no API key, run all checkov policies and all local custom policies
* with API key and without the `--include...` flag, run all policies that exist in the platform, all platform custom policies, and all local custom policies
* you can also `--check` specific check IDs to include them if they are checkov-only, without including all checkov policies
* with an API key and `--include...`, then runs all policies (same as the current behavior before this change)

### CODE REVIEW

**Review by kartikp10 — APPROVED** [review]
Noice 🚀

File: checkov/main.py
Comment (gruebel) [inline checkov/main.py:511]: nice 👍

File: checkov/main.py
Comment (gruebel) [inline checkov/main.py:342]:
```suggestion
               help='When running with an API key, Checkov will omit any policies that do not exist '
```
can be wrong, but sounds better without it 😄

**Review by gruebel — APPROVED** [review]
🍻

File: checkov/main.py
Comment (mikeurbanski1) [reply checkov/main.py:342]: You are absolutely not wrong, lol

**Comment by mikeurbanski1** [comment]
Wait until Sunday / Monday to merge, with a release announcement

**Comment by mikeurbanski1** [comment]
Merge on Sunday / Monday in conjunction with a release announcement

**Review by nimrodkor — COMMENTED** [review]
@mikeurbanski1 I don't think graph checks are affected by your changes... Changes should be around this method:
https://github.com/bridgecrewio/checkov/blob/d8ac7646a5d02173af3437bf1f31333b6cec6f2e/checkov/common/checks_infra/registry.py#L31

**Comment by mikeurbanski1** [comment]
> @mikeurbanski1 I don't think graph checks are affected by your changes... Changes should be around this method:
> 
> https://github.com/bridgecrewio/checkov/blob/d8ac7646a5d02173af3437bf1f31333b6cec6f2e/checkov/common/checks_infra/registry.py#L31

The metadata integration loads the graph checks before setting the metadata. I do see that it still only loads the TF graph check registry. It's fine up until now, because we have no OOTB graph checks for the others. Updated.

**Pull request merged**
